src/lib/definitions/eventdef.ts

```diff
@@ -14,11 +14,21 @@
  * limitations under the License.
  */
 import { Metadata } from './metadata';
-import { normalizeDataOnly, normalizeKind, overwriteCorrelation, overwriteMetadata } from './utils';
+import {
+  cleanSourceModelProperty,
+  normalizeDataOnly,
+  normalizeKind,
+  overwriteCorrelation,
+  overwriteMetadata,
+} from './utils';
 import { CorrelationDefs } from './types';
 
 export class Eventdef {
+  sourceModel?: Eventdef;
```

Comment:
The main change is this. I have created new property `sourceModel` to store the original value and removed it in the normalize method https://github.com/serverlessworkflow/sdk-typescript/pull/156/files#diff-d777c6a12a20ec15dd1884f631d697c0a39b4ac3d6ed1344786a33d6628325d5R82 before serialization

Having the source stored we can infer if the property has been set by the user or by us in the constructor.  https://github.com/serverlessworkflow/sdk-typescript/pull/156/files#diff-85a746dfd50b2ed38f2af0732b08ec5e2c76325048e9d6ae857a12120cda0361R708